.gitignore

```diff
@@ -1,2 +1,266 @@
-*.class
-*.log
+# Created by https://www.gitignore.io/api/osx,vim,code,linux,emacs,windows,eclipse,intellij+all
```

Comment:
Perhaps I'll add some comments with a bit of explanation.

My goal here was to ignore all system / OS-specific files (like OSX's `.DStore`) and editor / IDE config files which gets generated automatically and often accidentally committed.

I'm covering three main OS (Win, OSX, *nix) and common editors.

Reply:
When the `.gitignore` starts mentioning files that you didn't even know existed, there is a risk of accidentally not committing a file. And the pursuit of ignoring *all* system files is pointless because you never know what strange tools and temp files people are going to have. That's why Git has a user-specific `~/.gitignore` file.

Reply:
That makes sense... I'll revert.